include/sys/types.h

```diff
@@ -188,6 +188,10 @@ typedef intptr_t     ptrdiff_t;
 typedef uint16_t     wchar_t;
 #endif
 
+#if defined(__WCHAR_WIDTH__) && __WCHAR_WIDTH__ != 16
+#warning __WCHAR_WIDTH__ should be 16 on NuttX. Your toolchain might need -fshort-wchar.
```

Comment:
github show many warning happen with this change, all need be fixed before merge.

Reply:
yes. it's why this PR is a draft.